File: legacy/core/src/main/java/com/geometryduel/game/engine/PlayerEngine.java
package com.geometryduel.game.engine;

import com.geometryduel.game.actor.PlayerActor;

/** 还原 PlayerEngine：每帧向 inputDevice 写入操作意图。 */
public abstract class PlayerEngine {
    public float horizontalMove, verticalMove;
    public boolean shotButtonPressed, longShotButtonPressed, teleportButtonPressed;

    public abstract void run(PlayerActor player);

    public void operateMove(float x, float y) {
        this.horizontalMove = x;
        this.verticalMove = y;
    }

    public void operateMoveButton(int x, int y) {
        this.horizontalMove = x;
        this.verticalMove = y;
    }

    public void operateShotButton(boolean b) {
        this.shotButtonPressed = b;
    }

    public void operateLongShotButton(boolean b) {
        this.longShotButtonPressed = b;
    }

    public void operateTeleportButton(boolean b) {
        this.teleportButtonPressed = b;
    }
}

File: app/src/main/java/com/geometryduel/game/engine/PlayerEngine.kt
package com.geometryduel.game.engine

import com.geometryduel.game.actor.PlayerActor

/** 还原 PlayerEngine：每帧向 inputDevice 写入操作意图。 */
abstract class PlayerEngine {
    var horizontalMove = 0f
    var verticalMove = 0f
    var shotButtonPressed = false
    var longShotButtonPressed = false
    var teleportButtonPressed = false

    abstract fun run(player: PlayerActor)

    fun operateMove(x: Float, y: Float) {
        this.horizontalMove = x
        this.verticalMove = y
    }

    fun operateMoveButton(x: Int, y: Int) {
        this.horizontalMove = x.toFloat()
        this.verticalMove = y.toFloat()
    }

    fun operateShotButton(b: Boolean) {
        this.shotButtonPressed = b
    }

    fun operateLongShotButton(b: Boolean) {
        this.longShotButtonPressed = b
    }

    fun operateTeleportButton(b: Boolean) {
        this.teleportButtonPressed = b
    }
}
